pyscript.core/docs/README.md

```diff
@@ -0,0 +1,481 @@
+    print(value)
+
+sync.from_main = from_main
+```
+
+In the `xworker` counter part:
+
+```python
+# will log 1 and "two" in default stdout console
+xworker.sync.from_main(1, "two")
+```
+
+### The xworker global reference
+
+The content of the file used to initialize any `XWorker` on the main thread can always reach the `xworker` counter part as globally available (that means: no *import ... form ...* is necessary, it is already there).
+
+Within a *Worker* execution context, the `xworker` exposes the following features:
+
+| name          | example                                    | behavior |
+| :------------ | :------------------------------------------| :--------|
+| sync          | `xworker.sync.from_main(1, "two")`         | Executes the exposed `from_main` function in the main thread. Returns synchronously its result, if any. |
+| window        | `xworker.window.document.title = 'Worker'` | Differently from *pyodide* or *micropython* `import js`, this field allows every single possible operation directly in the main thread. It does not refer to the local `js` environment the interpreter might have decided to expose, it is a proxy to handle otherwise impossible operations in the main thread, such as manipulating the *DOM*, reading `localStorage` otherwise not available in workers, change location or anything else usually possible to do in the main thread. |
+| isWindowProxy | `xworker.isWindowProxy(ref)`               | **Advanced** - Allows introspection of *JS* references, helping differentiating between local worker references, and main thread global references. This is valid both for non primitive objects (array, dictionaries) as well as functions, as functions are also enabled via `xworker.window` in both ways: we can add a listener from the worker or invoke a function in the main. Please note that functions passed to the main thread will always be invoked asynchronously.
+
+```python
+print(xworker.window.document.title)
+
+xworker.window.document.body.append("Hello Main Thread")
+
+xworker.window.setTimeout(print, 100, "timers too")
+```
+
+ℹ️ - Please note that even if non blocking, if too many operations are orchestrated from a *worker*, instead of the *main* thread, the overall performance might still be slower due the communication channel and all the primitives involved in the synchronization process. Feel free to use the `window` feature as a great enabler for unthinkable or quick solutions but keep in mind it is still an indirection.
+
+#### The `sync` utility
+
+This helper does not interfere with the global context but it still ensure a function can be exposed form *main* and be used from *thread* and/or vice-versa.
+
+```python
+# main
+def alert_user(message):
+    import js
+    js.alert(message)
+
+w = XWorker('./file.py')
+# expose the function to the thread
+w.sync.alert_user = alert_user
+
+
+# thread
+if condition == None:
+    xworker.sync.alert_user('something wrong!')
+```
+
+
+## Custom Scripts
```

Comment:
Think the name "Scripts" here is a bit confusing (overloaded term).

Reply:
it's about having custom `<script type="...">` tags ... any suggestion?